bin/reth-bench/src/bench/context.rs

```diff
@@ -27,6 +30,13 @@ pub(crate) struct BenchContext {
     pub(crate) next_block: u64,
 }
 
+/// Client for retrieving and analyzing reth metrics
+/// 
+/// Should this be in its own file?
+pub(crate) struct MetricsClient {
+    pub(crate) metrics_url: Url,
+}
+
```

Comment:
If this is going to contain some methods for e.g. GETing the endpoint and doing things with the results, it would be good to be some `impl` methods for that, even if stubs